### ISSUE TITLE
next-i18next is acting weird

### ISSUE THREAD

**Issue by ktjd123** [isaachinman/next-i18next#887]
## Describe the bug
1. languages I added in other languages is not showing when I log with i18n.languages.
2. Sometimes load translation properly and show it on web, but sometimes It's just showing dev strings.

## Occurs in next-i18next version

    "i18next": "^19.8.4",
    "i18next-locize-backend": "^4.1.8",
    "next-i18next": "^7.0.1",

## Steps to reproduce
```js
const NextI18Next = require('next-i18next').default;
const path = require('path');

module.exports = new NextI18Next({
  otherLanguages: ['ko-KR', 'en', 'zh', 'zh-TW'],
  localePath: path.resolve('./public/static/locales'),
  cleanCode: true,
  fallbackLng: 'en',
  defaultLanguage: 'en',

  // Locize
  backend: {
    projectId: '8060db7d-5b16-49b9-82be-30cf650a7426',
    referenceLng: 'ko-KR',
  },
  use: [require('i18next-locize-backend/cjs')],
});
```
this is my i18n.js file. and added with @withTranslation() decorator. It acts same with HOC style.

```js
import { i18n as i, withTranslation } from '../i18n';
```
I used it from i18n.js file.

when I log with `console.log(navigator.language, i.language, i.languages);`
It logs `ko-KR en (1) ["en"]` and even prints the dev string with i.t() function.

## Expected behaviour
detect languages properly and show Translated strings.

## Screenshots
![스크린샷 2020-11-19 오후 11 36 17](https://user-images.githubusercontent.com/15044488/99680122-0e7b9680-2ac0-11eb-8599-a728466b75dd.png)
Screenshot from locize.

## OS (please complete the following information)
 - Device: MacBook Pro (15-inch, 2016), and server with linux acts same
 - Browser: Chrome, ios safari ... etc

## Additional context
It worked fine like week ago but it acts really weird. So I upgraded all dependencies but it acts the same.
I Always appreciate your works. Thank you.

**Comment by isaachinman** [comment]
You'll need to provide a reproducible repo. Also, why are you putting `en` both in `defaultLanguage` and in `otherLanguages`?

**Comment by ktjd123** [comment]
@isaachinman can i invite you to private repo? or make new one reproducing this issue? 

I didn't put it before but I thought the problem was this so I putted in there. Same when i remove it from `defaultLanguage`.

Thank you for fast reply

**Comment by isaachinman** [comment]
It should be a public repo, yes.

**Comment by ktjd123** [comment]
Can you check https://github.com/ktjd123/i18n-bug this repo, Thank you.

**Comment by isaachinman** [comment]
Hey @ktjd123, this looks like a fairly complex application including a custom server, etc. You will need to provide a _minimum_ reproduction case, so that we can understand where the problem is coming from.

**Comment by ktjd123** [comment]
Hi @isaachinman. I force pushed to same repo using next-i18next/example. Please recheck

**Comment by isaachinman** [comment]
@ktjd123 That app is now functioning exactly as I'd expect. If I set my browser to `ko-KR` and open in an incognito window, I see: `ko-KR ko-KR (2) ["ko-KR", "en"]`.

**Comment by ktjd123** [comment]
can you pull and check again? I updated the code.

![스크린샷 2020-11-20 오전 1 06 17](https://user-images.githubusercontent.com/15044488/99691560-b4cd9900-2acc-11eb-9cd3-036698ee50d1.png)
this is the second problem I said. t is not showing translated value.

with language detection problem. It sometimes show correctly but when I close the window and reconnect or restart server and reconnect, It does not work. I wasn't able to find out what is happening this.

**Comment by isaachinman** [comment]
As long as language detection is working, `next-i18next` is doing its job. It sounds like you probably have a problem with your `locize` backend.

**Comment by ktjd123** [comment]
This is screenshot for first bug.

![f](https://user-images.githubusercontent.com/15044488/99692823-12161a00-2ace-11eb-96fe-67fe0ecee24c.png)

I updated fallback for ko, and you can check in repo.

```ts
const NextI18Next = require('next-i18next').default;
const path = require('path');

module.exports = new NextI18Next({
  otherLanguages: ['ko-KR', 'en', 'zh', 'zh-TW'],
  localePath: path.resolve('./public/static/locales'),
  cleanCode: true,
  fallbackLng: {
    ko: ['ko-KR'],
    default: ['en'],
  },

  // Locize
  backend: {
    projectId: '8060db7d-5b16-49b9-82be-30cf650a7426',
    referenceLng: 'ko-KR',
  },
  use: [require('i18next-locize-backend/cjs')],
});
```

**Comment by ktjd123** [comment]
I think it's next-i18next's problem since it loads well from locize.

please check screenshot
![스크린샷 2020-11-20 오전 2 00 00](https://user-images.githubusercontent.com/15044488/99698503-3674f500-2ad4-11eb-8e39-b1ef22c6e9d9.png)

It loads then cannot find missing keys.

**Comment by ktjd123** [comment]
I think it's next-i18next's bug since it shows translated string when I call `i.changeLanguage('ko-KR')` function.

but on server log

```js
i18next::backendConnector: loaded namespace translation for language en {}
i18next::backendConnector: loaded namespace translation for language ko-KR {}
```
the log shows this.

**Comment by ktjd123** [comment]
second problem is solved by adding `ns: 'common'`.

defaultNs is set to common but ns was set to ['translation']. I wonder if this is bug

**Comment by ktjd123** [comment]
@isaachinman Can you check this bug in repo?

**Comment by isaachinman** [comment]
Hi there, I have just released `next-i18next@8.0.0`, which uses the internationalised routing built into NextJs v10+. This new version of `next-i18next` is quite different, thus making this PR no longer relevant. Thanks!

**Comment by adrai** [comment]
@ktjd123 there is an example with the new next-i18next version in combination with locize, here: https://github.com/locize/next-i18next-locize
